atlasdb-config/src/main/java/com/palantir/atlasdb/factory/startup/ConsistencyCheckRunner.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.factory.startup;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.palantir.async.initializer.AsyncInitializer;
+import com.palantir.atlasdb.factory.TransactionManagerConsistencyResult;
+import com.palantir.atlasdb.transaction.impl.consistency.TransactionManagerConsistencyCheck;
+import com.palantir.common.remoting.ServiceNotAvailableException;
+
+public final class ConsistencyCheckRunner extends AsyncInitializer {
+
+    private static final Logger log = LoggerFactory.getLogger(ConsistencyCheckRunner.class);
+
+    private static final RuntimeException UNKNOWN = new RuntimeException("unknown");;
+
+    private final List<TransactionManagerConsistencyCheck> consistencyChecks;
+
+    public ConsistencyCheckRunner(List<TransactionManagerConsistencyCheck> consistencyChecks) {
+        this.consistencyChecks = consistencyChecks;
+    }
+
+    @Override
+    protected void tryInitialize() {
+        TransactionManagerConsistencyResult consistencyResult = consistencyChecks.stream()
+                .map(Supplier::get)
+                .sorted(Comparator.comparing(
```

Comment:
Feels gross to depend on the ordering here. If we do this, leave a comment on the enum so that this doesn't change.

Reply:
You could maybe have a severity field on the enum which is a number and sort by that.

Reply:
Yep, will add severity. (Separately, the comparator should have been `reversed()` too - the old code would accidentally have picked the healthiest status, which is really bad!)

Reply:
Agreed, will add severity.